src/main/java/frc/robot/subsystems/TrapElvSubsystem.java

```diff
@@ -0,0 +1,374 @@
+        },
+        () -> {
+          stowTrapElv();
+        });
+  }
+
+  public Command scoreTrap() {
+    return startEnd(
+        () -> {
+          setTrapArm(TrapElvState.TRAP_SCORE);
+          rollerMotor.set(TrapElvConstants.rollerIntakeSpeed);
+        },
+        () -> {
+          stowTrapElv();
+        });
+  }
+
+  public void stowTrapElv() {
+    setTrapArm(TrapElvState.STOWED);
+    rollerMotor.stopMotor();
+  }
+
+  public Command zeroArm() {
+    return startEnd(
+        () -> {
+          // Command for zering elevator if elevator happens to be not at zero
+          // Runs elevator motors until there limit switches are pressed
+          if (!baseLimit.get()) {
+            baseMotor1.set(TrapElvConstants.elvZeroingSpeed);
+            baseMotor2.set(TrapElvConstants.elvZeroingSpeed);
+          } else {
+            baseMotor1.stopMotor();
+            baseMotor2.stopMotor();
+          }
+          if (!scoringLimit.get()) {
+            scoringMotor.set(TrapElvConstants.elvZeroingSpeed);
+          } else {
+            scoringMotor.stopMotor();
+          }
+        },
+        () -> {
+          baseMotor1.stopMotor();
+          baseMotor2.stopMotor();
+          scoringMotor.stopMotor();
+        });
+  }
+
+  public void setTrapArm(TrapElvState state) {
+    baseGoal.setDouble(Units.metersToInches(TrapElvConstants.elevatorMinHight) + state.basePose);
+    wristMotor.getPIDController().setReference(state.getWristPose(), ControlType.kPosition);
+    baseMotor1.getPIDController().setReference(state.getBasePose(), ControlType.kPosition);
+    baseMotor2.getPIDController().setReference(state.getBasePose(), ControlType.kPosition);
+    scoringMotor.getPIDController().setReference(state.getScoringPose(), ControlType.kPosition);
+  }
+
+  @Override
+  public void periodic() {
+    sourceLog.log(sourceBreak.get());
+    groundLog.log(groundBreak.get());
+    baseLog.log(baseLimit.get());
+    scoringLog.log(sourceBreak.get());
+  }
+
+  @Override
+  public void simulationPeriodic() {
+    for (double i = 0; i < Robot.defaultPeriodSecs; i += CANSparkMaxSim.kPeriod) {
+      m_baseElevatorSim.setInput(baseMotor1.get() * RobotController.getBatteryVoltage());
+      m_baseElevatorSim.update(CANSparkMaxSim.kPeriod);
+      baseMotor1.update(
+          m_baseElevatorSim.getVelocityMetersPerSecond() * 70 / Units.inchesToMeters(1));
+
+      m_scoringElevatorSim.setInput(scoringMotor.get() * RobotController.getBatteryVoltage());
+      m_scoringElevatorSim.update(CANSparkMaxSim.kPeriod);
+      scoringMotor.update(
+          m_scoringElevatorSim.getVelocityMetersPerSecond()
+              * TrapElvConstants.elevatorGearRatio
+              / Units.inchesToMeters(1));
+
+      m_wristMotorSim.setInput(wristMotor.get() * RobotController.getBatteryVoltage());
+      m_wristMotorSim.update(CANSparkMaxSim.kPeriod);
+      wristMotor.update(m_wristMotorSim.getVelocityRadPerSec());
```

Comment:
Gear ratio needs to be factored in here I think. 

Also, I think the plan is to use the CANCoders for the wrist, so we have to use the PIDController on the roboRIO and can't integrate with the SparkMax controller. Okay to address this in a future PR though (just make a new work item so we don't forget).